### DISCUSSION TITLE
Can DataModels be used without a remote data source?

### DISCUSSION THREAD

**Discussion by wardy484** [flutterdata/flutter_data#201] [Q&A] [answered]
I've tried generating a model with no remote repository and I've written some rows to the database something I'm noticing is that the records that have persisted so far do not yet have an ID. 

I have some data I would like to keep local such settings or downloads specific to a device, but I was hoping to be able to use flutter data for all data access. So I just wanted to confirm if my use case is normal and if so perhaps I'm missing something?

😅 is it as simple as just managing local record IDs myself?

**Comment by frank06** [comment, marked as answer]
Yes!  I like to use `@DataRepository([], remote: false)` to make all calls local by default.

**Reply by wardy484** [reply]
Great, that's good to know. I hadn't spotted that in the docs. 

One more question if you don't mind. Will watchers still refresh if the database is updated in another process, i.e. an isolate generated by work_manager.

**Reply by frank06** [reply]
Watchers get triggered by certain internal actions in FD, not directly by changes in the database. I think we would need to migrate to Isar and make some other changes to have proper cross-isolate support.

**Reply by wardy484** [reply]
That makes sense. 😅 I've currently got a mix of firestore and isar in my app and I'd love to get rid of firestore 😮‍💨  Isar has served me really well. I didn't even realize that cross-isolate support was going to be issue until I tried drift. 

I saw an issue for migration to Isar but the difference between Isar and Hive seem quite significant so I'm guessing it's not a small feat to migrate.

**Reply by frank06** [reply]
> I'm guessing it's not a small feat to migrate.

You're absolutely right.  I wish I had more time (funding?) for that.

I hear you, Firebase libraries are terrible. We (in a big app) unfortunately had to rewrite Firestore and Firebase Storage SDKs to hit the APIs directly from Flutter Data adapters. And yes, my code is working much better than Google's. 😅
